### PR TITLE
Media gpu priority setting

### PR DESCRIPTION
**Pull request by hongbolv**

### CODE REVIEW

File: media_driver/linux/common/os/mos_context_specific.h [hunk 1]
```diff
@@ -202,6 +202,19 @@ class OsContextSpecific : public OsContext
     void SetSliceCount(uint32_t *pSliceCount);
 
 #endif
+    //!
+    //! \brief  Get the context priority from KMD
+    //! \param  [in, out] pPriority
+    //!         Pointer to the priority of current gpu context.
+    //!
+    void GetGpuPriority(int32_t *pPriority);
```
Comment (Xiaogangli-intel) [inline media_driver/linux/common/os/mos_context_specific.h:210]: Which platform are these functions for?
Maybe we also need to add these functions in OsContextSpecificNext.

File: media_driver/linux/common/codec/ddi/media_libva_encoder.cpp [hunk 2]
```diff
@@ -378,8 +380,11 @@ VAStatus DdiEncode_CreateContext(
 
     MOS_STATUS eStatus = pCodecHal->Allocate(encCtx->m_encode->m_codechalSettings);
 
-#ifdef _MMC_SUPPORTED
+    //Get the priority from VAAPI.
```
Comment (Xiaogangli-intel) [inline media_driver/linux/common/codec/ddi/media_libva_encoder.cpp:383]: should be **_set_** the priority, right?

File: media_driver/linux/common/codec/ddi/media_libva_encoder.cpp [hunk 3]
```diff
@@ -378,8 +380,11 @@ VAStatus DdiEncode_CreateContext(
...
     PMOS_INTERFACE osInterface = pCodecHal->GetOsInterface();
+    osInterface->pfnSetGpuPriority(osInterface, priority);
```
Comment (Xiaogangli-intel) [inline media_driver/linux/common/codec/ddi/media_libva_encoder.cpp:385]: Please check osInterface is available before using it.

File: media_driver/linux/common/ddi/media_libva_caps.cpp [hunk 4]
```diff
@@ -574,6 +584,24 @@ VAStatus MediaLibvaCaps::CheckAttribList(
...
 
+VAStatus MediaLibvaCaps::GetGeneralConfigAttrib(VAConfigAttrib* attrib)
+{
+    DDI_CHK_NULL(attrib, "Null pointer", VA_STATUS_ERROR_INVALID_PARAMETER);
+
+    VAStatus status = VA_STATUS_SUCCESS;
+#if VA_CHECK_VERSION(1, 9, 0)
+    if (attrib->type == VAConfigAttribContextPriority)
+    {
+        attrib->value = CONTEXT_PRIORITY_MAX;
```
Comment (Xiaogangli-intel) [inline media_driver/linux/common/ddi/media_libva_caps.cpp:595]: So, what's the default value? CONTEXT_PRIORITY_MAX or CONTEXT_PRIORITY_MAX/2?
I saw set the default value as CONTEXT_PRIORITY_MAX/2 at other places.

Comment (hongbolv) [reply media_driver/linux/common/ddi/media_libva_caps.cpp:595]: The code is to report the max priority to upper application. It should be CONTEXT_PRIORITY_MAX

File: media_driver/linux/common/codec/ddi/media_libva_encoder.cpp (hunk 3 above)
Comment (hongbolv) [reply media_driver/linux/common/codec/ddi/media_libva_encoder.cpp:391]: Done

File: media_driver/linux/common/codec/ddi/media_libva_encoder.cpp (hunk 2 above)
Comment (hongbolv) [reply media_driver/linux/common/codec/ddi/media_libva_encoder.cpp]: Done.

File: media_driver/linux/common/os/mos_context_specific.h (hunk 1 above)
Comment (hongbolv) [reply media_driver/linux/common/os/mos_context_specific.h:210]: It is a general feature. It is validated on Gen12 TGL.

File: media_driver/linux/common/ddi/media_libva_caps.cpp [hunk 5]
```diff
@@ -2420,18 +2505,24 @@ VAStatus MediaLibvaCaps::GetDecConfigAttr(
...
         VAConfigID configId,
         VAProfile *profile,
-        VAEntrypoint *entrypoint)
+        VAEntrypoint *entrypoint,
+        int32_t *priority)
 {
     DDI_CHK_NULL(profile, "Null pointer", VA_STATUS_ERROR_INVALID_PARAMETER);
     DDI_CHK_NULL(entrypoint, "Null pointer", VA_STATUS_ERROR_INVALID_PARAMETER);
     int32_t profileTableIdx = -1;
-    int32_t configOffset = configId - DDI_VP_GEN_CONFIG_ATTRIBUTES_BASE;
```
[the remark this answers is not in the document]
Comment (hongbolv) [reply media_driver/linux/common/ddi/media_libva_caps.cpp:2434]: It is an existing issue. In the next function call,  the "- DDI_VP_GEN_CONFIG_ATTRIBUTES_BASE" is executed again.

**Comment by hongbolv** [comment]
Changed to dynamic session priority implementation.

**Pull request closed without merging**